docs/package_templates/header_only/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
+import os
+
+
+required_conan_version = ">=1.51.3"
+
+
+class PackageConan(ConanFile):
+    name = "package"
+    description = "short description"
+    license = "" # Use short name only, conform to SPDX License List: https://spdx.org/licenses/
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/project/package"
+    topics = ("topic1", "topic2", "topic3", "header-only") # no "conan"  and project name in topics, keep 'header-only'
+    settings = "os", "arch", "compiler", "build_type" # even for header only
+    no_copy_source = True # do not copy sources to build folder for header only projects
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "14.1",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+
+    # same package ID for any package
+    def package_id(self):
+        self.info.clear()
+
+    def requirements(self):
+        # prefer self.requires method instead of requires attribute
+        self.requires("dependency/0.8.1")
+
+    # Only in case the project is header-only
+    def package_id(self):
+        self.info.clear()
+
+    # if another tool than the compiler or CMake is required to build the project (pkgconf, bison, flex etc)
+    def build_requirements(self):
+        self.tool_requires("tool/x.y.z")
+
+    def source(self):
+        # download source package and extract to source folder
+        get(**self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def validate(self):
+        # validate the minimum cpp standard supported
+        if self.info.settings.compiler.cppstd:
```

Comment:
AFAIK, doesn't work yet if there `self.info.clear()` in `package_id()`

Reply:
True, I'll remove. The point is, on Conan 2.0 cppstd will be always expected.

Reply:
I've updated to `self.info.settings.get_safe("compiler.cppstd")`. compiler is None when building, but when installing, it will be used to validate.